examples/wifi_static_ip.rs

```diff
@@ -0,0 +1,111 @@
+//! Example of using blocking wifi with a static IP configuration
+//!
+//! Add your own ssid and password for the access point
+//! Add your own gateway IP, netmask, and local device IP for interface configuration
+
+use embedded_svc::wifi::{AuthMethod, ClientConfiguration, Configuration as WifiConfiguration};
+use std::net::Ipv4Addr;
+use std::str::FromStr;
+
+use esp_idf_svc::hal::prelude::Peripherals;
+use esp_idf_svc::ipv4::{
+    ClientConfiguration as IpClientConfiguration, ClientSettings as IpClientSettings,
+    Configuration as IpConfiguration, Mask, Subnet,
+};
+use esp_idf_svc::log::EspLogger;
+use esp_idf_svc::netif::{EspNetif, NetifConfiguration, NetifStack};
+use esp_idf_svc::wifi::{BlockingWifi, EspWifi};
+use esp_idf_svc::{eventloop::EspSystemEventLoop, nvs::EspDefaultNvsPartition};
+
+use log::info;
+
+const SSID: &str = env!("WIFI_SSID");
+const PASSWORD: &str = env!("WIFI_PASS");
+
+// Expects IPv4 address
+const DEVICE_IP: &str = env!("ESP_DEVICE_IP");
+// Expects IPv4 address
+const GATEWAY_IP: &str = env!("GATEWAY_IP");
+// Expects a number between 0 and 32, defaults to 24
+const GATEWAY_NETMASK: Option<&str> = option_env!("GATEWAY_NETMASK");
+
+fn main() -> anyhow::Result<()> {
+    esp_idf_svc::sys::link_patches();
+    EspLogger::initialize_default();
+
+    let peripherals = Peripherals::take()?;
+    let sys_loop = EspSystemEventLoop::take()?;
+    let nvs = EspDefaultNvsPartition::take()?;
+
+    let mut wifi = EspWifi::new(peripherals.modem, sys_loop.clone(), Some(nvs))?;
+    configure_wifi(&mut wifi);
```

Comment:
I think it would be much clearer if - rather than configuring `EspWifi` with default settings - and then - immediately swapping out the configured STA (and AP) netifs, you instead just **create** the `EspWifi` instance with your custom STA *from the very beginning*.

I.e. something like this (excuse potential typos):
```rust
let mut wifi = EspWifi::wrap_all(
    WifiDriver::new(peripherals.modem, sys_loop.clone(), Some(nvs))?,
    EspNetif::new_with_conf(&NetifConfiguration {
        ip_configuration: IpConfiguration::Client(IpClientConfiguration::Fixed(IpClientSettings {
            ip: static_ip,
            subnet: Subnet {
                gateway: gateway_addr,
                mask: Mask(netmask),
            },
            // Can also be set to Ipv4Addrs if you need DNS
            dns: None,
            secondary_dns: None,
        })),
        ..NetifStack::Sta::default_configuration()
    }),
    EspNetif::new(NetifStack::Ap),
)?;
```

Reply:
Thank you for pointing me in the right direction, that is much more sensible!